homeassistant/components/group/sensor.py

```diff
@@ -444,3 +458,45 @@ def _calculate_entity_properties(self) -> None:
             and all(x == unit_of_measurements[0] for x in unit_of_measurements)
         ):
             self.calc_unit_of_measurement = unit_of_measurements[0]
+
+        if self.entities_can_convert:
+            (
+                _,
+                _,
+                self.calc_unit_of_measurement,
+            ) = self._check_entities_device_class_and_uom()
+
+    def _check_entities_device_class_and_uom(
+        self,
+    ) -> tuple[bool, type[BaseUnitConverter] | None, str | None]:
+        """See if we can convert values if device_classes and UoM matches."""
+
+        _device_classes: list[str] = []
+        entity_reg = er.async_get(self.hass)
+        for entity_id in self._entity_ids:
+            entity = entity_reg.async_get(entity_id)
+            if not entity or not entity.device_class:
+                return False, None, None
+            _device_classes.append(entity.device_class)
+        if not all(x == _device_classes[0] for x in _device_classes):
+            return False, None, None
+
+        first_device_class = SensorDeviceClass(_device_classes[0])
```

Comment:
This will raise `ValueError` if the sensor is using a string device class which does not match any `SensorDeviceClass`, that needs to be handled.
There's a helper, `try_parse_enum`, which returns `None` in case that suits the code better.